crates/sui-sdk-crypto/src/lib.rs

```diff
@@ -0,0 +1,41 @@
+#![cfg_attr(doc_cfg, feature(doc_cfg))]
+
+use sui_sdk::types::PersonalMessage;
+use sui_sdk::types::Transaction;
+use sui_sdk::types::UserSignature;
+
+pub use signature::Error as SignatureError;
+pub use signature::Signer;
+pub use signature::Verifier;
+
+#[cfg(feature = "ed25519")]
+#[cfg_attr(doc_cfg, doc(cfg(feature = "ed25519")))]
+pub mod ed25519;
+
+#[allow(unused)]
+mod bls12381;
+#[allow(unused)]
+mod secp256k1;
+#[allow(unused)]
+mod secp256r1;
+
+pub trait SuiSigner {
+    fn sign_transaction(&self, transaction: &Transaction) -> Result<UserSignature, SignatureError>;
+    fn sign_personal_message(
+        &self,
+        message: &PersonalMessage<'_>,
+    ) -> Result<UserSignature, SignatureError>;
+}
```

Comment:
Do public keys normalize somehow? or would it be good to leave each pubkey() call out of the trait.

Reply:
can each impl of SuiSigner return the same type for `public_key()` or should they just be kept separate. On chain each address resolves to some `0x123...` which might be useful when using a signer as a common function.

Reply:
So a signing API doesn't necessarily need to provide its public key info (Every signature in Sui will curry this information anyway). So are you more interested in ensuring that you can take a signer and easily ask/get the sui address that it corresponds to or do you want the pubkey info as it would possibly appear in a multisig address?

Reply:
Either way can be fine, it just might be nice to identify which signer you currently have. It's a common function in other signer interfaces.

Reply:
yeah of course, we can definitely expose that info